src/java.base/share/classes/java/lang/AbstractStringBuilder.java

```diff
@@ -468,9 +468,7 @@ public int codePointCount(int beginIndex, int endIndex) {
      *   {@code codePointOffset} code points.
      */
     public int offsetByCodePoints(int index, int codePointOffset) {
-        if (index < 0 || index > count) {
-            throw new IndexOutOfBoundsException();
-        }
+        checkOffset(index, count);
```

Comment:
String.offsetByCodePoints is specified to throw IOOBE. checkOffset may throw the more specific StringIndexOutOfBoundsException. That's a compatible change but I worry that we might want to throw the less specific exception in the further because code. I only mention is because there have been cases in java.lang where IOOBE was specified and AIOOBE by the implementation and it has been problematic to touch the implementation as a result.

Reply:
Yes, changing the type of thrown exception may break something. And as David said, this also requires a CSR approval, which is a relatively long process, so I restored the original code.